R/repair-names.R

```diff
@@ -172,12 +172,14 @@ repaired_names <- function(name,
   new_name
 }
 
-minimal_names <- function(name, n) {
+minimal_names <- function(name, n = NULL) {
   if (is.null(name) && missing(n)) {
     abort(error_name_length_required())
   }
-  ## TODO: address scenarios where name is not NULL and n != length(name)?
   new_name <- name %||% rep_len("", n)
+  if (!is.null(n)) {
+    length(new_name) <- n
+  }
```

Comment:
It feels like `minimal_names()` should gain a test or two exploring cases where `n` is not equal to `length(name)`.

Reply:
What do you think of being more strict, i.e. `n` is only consulted when `is.null(name)`? I don't see a reason to make `minimal_names()` capable of changing the length of a non-`NULL` `name`.

``` r
devtools::load_all("~/rrr/tibble")
#> Loading tibble

x <- c("a", "", NA)

minimal_names(x)
#> [1] "a" ""  ""

minimal_names(x, n = 2)
#> [1] "a" ""

minimal_names(x, n = 5)
#> [1] "a" ""  ""  ""  ""
```

<sup>Created on 2018-10-19 by the [reprex package](https://reprex.tidyverse.org) (v0.2.1)</sup>